### DISCUSSION TITLE
Undo completion when there is only one match

### DISCUSSION THREAD

**Discussion by ImNotaGit** [marlonrichert/zsh-autocomplete#656] [Q&A]
Hi, thanks for this wonderful plugin. I have this plugin set up as follows:

```
# this is to make zsh-autocomplete not show autocomplete menu unless the user triggers it with Tab, etc.; this needs to be before sourcing zsh-autocomplete
zstyle ':autocomplete:async' enabled no
# make Tab go directly to completion menu and cycle there
bindkey '\t' menu-select "$terminfo[kcbt]" menu-select
bindkey -M menuselect '\t' menu-complete "$terminfo[kcbt]" reverse-menu-complete
# Enter to directly execute command
bindkey -M menuselect '\r' .accept-line
# Esc to abort
bindkey -M menuselect '^[' send-break
# Backspace to undo completion
bindkey -M menuselect '^?' undo
```

With this, when a partial command I type has multiple matches, I could `<Tab>` through the different matches, and when I press `<Backspace>` (or `<Esc>`), due to my settings it will undo the last match and restore the command line to the state where I last explicitly typed anything. However, when there is only one match, it seems that the match was automatically accepted, and the undo doesn't work. I wonder whether there is any way to make undo work even with only one match. This is because sometimes I make a typo that results in a single exact match that is not what I want (and sometimes the match is from recent file and can be quite long), and then I had to painstakingly delete the completed text.

Thanks again for your help.

**Comment by marlonrichert** [comment]
It doesn't work because `undo` does not have the same keybinding on the command line:
```sh
% PS1='%# ' zsh -f
% bindkey | grep undo
"^X^U" undo
"^Xu" undo
"^_" undo
%
```

If you want this to work intuitively, then you need to use the same binding in the menu as you have on the command line.
